### PR TITLE
manywheel: Move to manylinux2014

### PR DESCRIPTION
**Pull request by seemethere**
Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

### CODE REVIEW

File: manywheel/Dockerfile [hunk 1]
```diff
@@ -1,17 +1,44 @@
...
+        perl \
+        unzip \
+        util-linux \
+        wget \
+        which \
+        xz \
+        yasm
 RUN yum install -y yum-utils centos-release-scl
 RUN yum-config-manager --enable rhel-server-rhscl-7-rpms
 RUN yum install -y devtoolset-7-gcc devtoolset-7-gcc-c++ devtoolset-7-gcc-gfortran devtoolset-7-binutils
```
Comment (seemethere) [inline manywheel/Dockerfile:39]: We should also probably have a discussion on whether or not we want to use devtoolset-7 or devtoolset-9, which is what manylinux2014 upstream uses 

https://github.com/pypa/manylinux/blob/bb4aa00ea89acf22d3b35c6d909660da84533b31/docker/build_scripts/build.sh#L63

Comment (soumith) [reply manywheel/Dockerfile:39]: dont care as long as it compiles. afaicr devtoolset9 (gcc-9) didn't compile pytorch when we checked a long time ago. it probably does, now.

Comment (seemethere) [reply manywheel/Dockerfile:39]: This is actually still the case, the highest version of gcc that CUDA 10.1 supports is still `8.x` :(

Comment (seemethere) [reply manywheel/Dockerfile:39]: Do you think it'd be an issue if the version of python we compile against was compiled with devtoolset-9 while we still compile with devtoolset-7?

Comment (mattip) [reply manywheel/Dockerfile:39]: I think it is fine. Linux does not suffer from the same runtime-mixing problems that windows has.

**Comment by janaknat** [comment]
Will this PR be accepted? We are looking at add support for pytorch aarch64 wheels. That will require manylinux2014.

**Comment by seemethere** [comment]
I'm currently re-doing this within the pytorch repository here: https://github.com/pytorch/pytorch/pull/45091

**Pull request closed without merging**